### DISCUSSION TITLE
Node.js and copyleft licenses

### DISCUSSION THREAD

**Discussion by thernstig** [nodejs/help#4950] [Q&A] [answered]
**TLDR:** Does projects using Node.js fall under copyleft licenses, such as GPL?

Looking at https://github.com/nodejs/node/blob/main/LICENSE it states that some parts of external libraries use *GNU General Public License (GPL)*. These are copyleft licenses.

As an example, check https://github.com/nodejs/node/blob/5fad0b93667ffc6e4def52996b9529ac99b26319/deps/uv/m4/libuv-check-flags.m4 (libuv is listed as a library Node.js depends on, see https://nodejs.org/en/docs/meta/topics/dependencies/).

So does this mean anyone using Node.js has to share their source code with anyone?

**Comment by Trott** [comment, marked as answer]
> So does this mean anyone using Node.js has to share their source code with anyone?

Standard disclaimer: I am not a lawyer and this is not legal advice.

**No, using Node.js does not mean you need to distribute your source code.**

However, if you are sharing/selling/distributing a modified version of the Node.js executable, it's possible that you need to make the source code for your modified version of Node.js available due to the terms of the GPL components within Node.js. I don't actually know. Maybe you do. Maybe you don't. Maybe it depends on a lot of factors.

But for using Node.js or distributing unmodified copies of Node.js, the GPL does not restrict that. You are free to use Node.js without making your source code available.

**Reply by thernstig** [reply]
@Trott great answer, thanks. Should this be somewhere in the Node.js docs you believe, or is it expected users to know this by heart? Both knowing Node.js has copyleft licenses, and what it entails, is most likely something most do not know.
